### PR TITLE
Feature/324 add api description for json functions

### LINKED ISSUE #276: Improvements for API documentation
## API
I found a few inaccuracies in the Virtual Schema API Documentation while using it.

Missing literal_interval json example here: https://github.com/exasol/virtual-schemas/blob/master/doc/development/virtual_schema_api.md#literal

I discovered that "type": "predicate_equals" doesn't really work in our implementation, but "type": "predicate_equal" works fine. We need to fix the documentation or the implementation. You can find it here: https://github.com/exasol/virtual-schemas/blob/master/doc/development/virtual_schema_api.md#predicates

The same for "predicate_notequals".

The same for "predicate_lessequals".

Missing predicate_is_null and predicate_is_not_null here: https://github.com/exasol/virtual-schemas/blob/master/doc/development/virtual_schema_api.md#predicates

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #276 #260 #324 #327

### CODE REVIEW

File: doc/development/developing-sql-dialect/implementing_additional_dialect_specific_behavior.md [hunk 1]
```diff
@@ -235,11 +235,11 @@ That way the Exasol database can directly run `IMPORT` on the remote data source
 
 ###  Pre-Requisites for Using IMPORT
 
-If you plan to use `IMPORT`, bare in mind that remote data source must **offer a JDBC driver** (always the case if you write a dialect for the JDBC adapter instead of an adapter from scratch).
+If you plan to use `IMPORT`, bear in mind that remote data source must **offer a JDBC driver** (always the case if you write a dialect for the JDBC adapter instead of an adapter from scratch).
 
 ### Overloading Rewriting
 
-The `AbstractSqlDialect` has a base implementation for query rewriting that should work with most databases that support JDBC. 
+The `AbstractSqlDialect` has a base implementation for the query rewriting that should work with most databases that support JDBC.
```
Comment (redcatbear) [inline doc/development/developing-sql-dialect/implementing_additional_dialect_specific_behavior.md:242]: "the" is not necessary here.

File: doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ And we also need two corresponding test classes:
     And one more **package for tests**:  
     `/virtual-schemas/src/test/java/com/exasol/adapter/dialects/<your new dialect package>` 
 
-    For example, the package for Athena is named `com.exasol.adapter.dialects.athena`.
+    For example, the package for Athena has a name `com.exasol.adapter.dialects.athena`.
```
Comment (redcatbear) [inline doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md:24]: Please revert this change. This sounds strange.

File: doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md [hunk 3]
```diff
@@ -42,18 +42,18 @@ And we also need two corresponding test classes:
    } 
    ```
 
-1. **Add a constructor** that takes a [JDBC database connection](https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html) and user properties as parameters. 
+1. **Add a constructor** that takes a [JDBC connection](https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html) Factory and user properties as parameters.
```
Comment (redcatbear) [inline doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md:45]:
```suggestion
1. **Add a constructor** that takes a [JDBC connection](https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html) factory and user properties as parameters. 
```

File: doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md [hunk 4]
```diff
@@ -42,18 +42,18 @@ And we also need two corresponding test classes:
    } 
    ```
 
-1. **Add a constructor** that takes a [JDBC database connection](https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html) and user properties as parameters. 
+1. **Add a constructor** that takes a [JDBC connection](https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html) Factory and user properties as parameters. 
     Your IDE could also generate it for you.
```
Comment (redcatbear) [inline doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md:46]:
```suggestion
    Or let your IDE generate it for you. You might wonder, why you get a factory for a connection instead of a connection. This allows your dialect to decide if and when you really need a connection to the remote data source. Depending on that source, establishing a connection can be costly, so having full control over connection creation can safe resources and execution time.
```

File: doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md [hunk 5]
```diff
@@ -144,10 +144,12 @@ And we also need two corresponding test classes:
     
     public class AthenaSqlDialectTest {
```
Comment (redcatbear) [inline doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md:145]:
```suggestion
    
    @ExtendWith(MockitoExtension.class)
    public class AthenaSqlDialectTest {
```

Also please add the necessary import.

File: doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md [hunk 6]
```diff
@@ -336,12 +338,12 @@ And we also need two corresponding test classes:
     ```java
     @Override
     protected RemoteMetadataReader createRemoteMetadataReader() {
-        return new BaseRemoteMetadataReader(this.connection, this.properties);
+        return new BaseRemoteMetadataReader(this.connectionFactory.getConnection(), this.properties);
```
Comment (redcatbear) [inline doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md:341]: I am guessing there is a `try { ... } catch { ... }` block missing here.

File: doc/user-guide/user_guide.md [hunk 7]
```diff
@@ -18,7 +18,8 @@ Before you can start using Virtual Schemas you should know:
 1. [Aurora](../dialects/aurora.md)
 1. [Big Query](../dialects/bigquery.md)
 1. [DB2](../dialects/db2.md)
-1. [EXASOL](../dialects/exasol.md)
+1. [Exasol](https://github.com/exasol/exasol-virtual-schema/blob/master/doc/dialects/exasol.md)
+**Attention!** The Exasol dialect was moved to a separate repository. If you need to use the Exasol dialect, find the latest release [here](https://github.com/exasol/exasol-virtual-schema/releases).
```
Comment (redcatbear) [inline doc/user-guide/user_guide.md:22]: Change according to my comment above.

File: doc/user-guide/user_guide.md [hunk 8]
```diff
@@ -60,7 +61,7 @@ Before you can start using the JDBC adapter for virtual schemas you have to depl
 
 ## Deploying JDBC Driver Files
 
-You have to upload the JDBC driver files of your remote database **twice** (except the BigQuery dialect):
+You have to upload the JDBC driver files of your remote database **twice** (except the Exasol and BigQuery dialects):
```
Comment (redcatbear) [inline doc/user-guide/user_guide.md:64]:
```suggestion
You have to upload the JDBC driver files of your remote database **twice** (except for the Exasol and BigQuery dialects):
```

File: doc/development/api/capabilities_list.md [hunk 9]
```diff
@@ -0,0 +1,268 @@
+# Capabilities List
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/development/api/capabilities_list.md:1]: I created a ticket for that:
https://github.com/exasol/virtual-schemas/issues/332

File: doc/development/developing-sql-dialect/implementing_additional_dialect_specific_behavior.md (hunk 1 above)
Comment (AnastasiiaSergienko) [reply doc/development/developing-sql-dialect/implementing_additional_dialect_specific_behavior.md:242]: My IDE's spell checker was complaining about it

File: doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md (hunk 2 above)
Comment (AnastasiiaSergienko) [reply doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md:24]: done

File: doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md (hunk 6 above)
Comment (AnastasiiaSergienko) [reply doc/development/developing-sql-dialect/implementing_mandatory_sql_dialect_classes.md:342]: done

File: doc/user-guide/user_guide.md (hunk 7 above)
Comment (AnastasiiaSergienko) [reply doc/user-guide/user_guide.md:22]: done

File: README.md [hunk 10]
```diff
@@ -118,18 +119,15 @@ Running the Virtual Schema requires a Java Runtime version 9 or later.
...
 | [JUnit](https://junit.org/junit5)                                                   | Unit testing framework                                 | Eclipse Public License 1.0    |
 | [J5SE](https://github.com/itsallcode/junit5-system-extensions)                      | JUnit5 extensions to test Java System.x functions      | Eclipse Public License 2.0    |
 | [Mockito](http://site.mockito.org/)                                                 | Mocking framework                                      | MIT License                   |
-| [SnakeYaml](https://bitbucket.org/asomov/snakeyaml/src/default/)                    | YAML parsing                                           | Apache License 2.0            |
+| [Testcontainers](https://www.testcontainers.org/)                                   | Container-based integration tests                      | MIT License                   |
+| JDBC driver(s), depending on data source                                            | Connecting to the data source for testing              | Check driver documentation    |
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply README.md:129]: done
